_docker/drupal/drupal-filesystem/web/modules/custom/rhd_purpose_attributes/src/Controller/PurposeAttributesPageController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\rhd_purpose_attributes\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Link;
+use Drupal\core\Url;
+
+/**
+ * Class PurposeAttributesPageController.
+ *
+ * @package Drupal\rhd_purpose_attributes\Controller
+ */
+class PurposeAttributesPageController extends ControllerBase {
+
+  /**
+   * Returns a purpose_attributes_report_block block.
+   *
+   * @return array
+   *   Returns a block render array.
+   */
+  public function renderBlockContent() {
+    $plugin_block = \Drupal::service('plugin.manager.block')
+      ->createInstance('purpose_attributes_report_block', []);
+    $access_result = $plugin_block->access(\Drupal::currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      return [];
+    }
+    $build = $plugin_block->build();
+    return $build;
+  }
+
+  /**
+   * Render the content.
+   */
+  public function content() {
+
+    $content = $this->renderBlockContent();
+
+    // Render link to csv output.
+    // @TODO: Get route to the view dynamically.
+    $link = Link::fromTextAndUrl('Download CSV', Url::fromUri('internal:/feed/property_export'));
```

Comment:
I was debating if it is worth grabbing the View by the display_id and getting the URL that way, but it seems to me like we will have to 'hard code' either the URL or some identifier in some way. I guess perhaps a machine name might provide slightly more flexibility, but I am also not sure why we would want to change this URL. I think that we can just keep that in the back of our mind just in case we need/want to be able to change the URL at some point down the road without requiring a code change.